src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -2,16 +2,59 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Security.AccessControl;
 using Xunit;
 
 namespace System.IO
 {
     public class FileSystemAclExtensionsTests
     {
         [Fact]
-        public void GetAccessControl_InvalidArguments()
+        public void GetAccessControl_DirectoryInfo_InvalidArguments()
         {
             Assert.Throws<NullReferenceException>(() => FileSystemAclExtensions.GetAccessControl((DirectoryInfo)null));
         }
+
+        [Fact]
+        public void GetAccessControl_DirectoryInfo_AccessControlSections_InvalidArguments()
+        {
+            Assert.Throws<NullReferenceException>(() => FileSystemAclExtensions.GetAccessControl((DirectoryInfo)null, new AccessControlSections()));
+        }
+
+        [Fact]
+        public void GetAccessControl_FileInfo_InvalidArguments()
+        {
+            Assert.Throws<NullReferenceException>(() => FileSystemAclExtensions.GetAccessControl((FileInfo)null));
+        }
+
+        [Fact]
+        public void GetAccessControl_FileInfo_AccessControlSections_InvalidArguments()
+        {
+            Assert.Throws<NullReferenceException>(() => FileSystemAclExtensions.GetAccessControl((FileInfo)null, new AccessControlSections()));
+        }
+
+        [Fact]
+        public void GetAccessControl_Filestream_InvalidArguments()
+        {
+            Assert.Throws<NullReferenceException>(() => FileSystemAclExtensions.GetAccessControl((FileStream)null));
+        }
+
+        [Fact]
+        public void SetAccessControl_DirectoryInfo_DirectorySecurity_InvalidArguments()
+        {
+            Assert.Throws<ArgumentNullException>(() => FileSystemAclExtensions.SetAccessControl((DirectoryInfo)null, (DirectorySecurity)null));
```

Comment:
This doesn't make it clear that both arguments are checked, because you pass null to both. Indeed looking at the code it seems that `directoryInfo` isn't checked which looks like a bug to me (but maybe there's some reason for it being as it is).

It's better to have just the one invalid argument in a test of validation, expect obviously when validation is a matter of relation between two or more arguments.

Also you can include a check on parameter names for exceptions derived from `ArgumentException`, so if you had:

```C#
Assert.Throws<ArgumentNullException>("directorySecurity", () => FileSystemAclExtensions.SetAccessControl(validDirectoryInfo, (DirectorySecurity)null));
```

Then it would also validate the parameter name in the exception.